src/js/components/FormField/FormField.js

```diff
@@ -190,14 +194,57 @@ const FormField = forwardRef(
       pad,
       required,
       style,
-      validate,
+      validate: validateProp,
       validateOn,
       ...rest
     },
     ref,
   ) => {
+    const validate = isObject(validateProp)
+      ? { ...validateProp }
+      : validateProp;
     const theme = useContext(ThemeContext) || defaultProps.theme;
     const formContext = useContext(FormContext);
+    const { format } = useContext(MessageContext);
```

Comment:
Some of the complexity in this and how it's having to pull in other dependencies make me think this should be done in Form.js validate() instead where the required and general function are done